File: Lab9/app/src/main/kotlin/com/example/currencyexchange/api/model/Transaction.kt
package com.example.currencyexchange.api.model

import com.google.gson.annotations.SerializedName

data class Transaction(
    @SerializedName("usd_amount") val usdAmount: Float,
    @SerializedName("lbp_amount") val lbpAmount: Float,
    @SerializedName("usd_to_lbp") val usdToLbp: Boolean,
    @SerializedName("id") val id: Int? = null,
    @SerializedName("added_date") val addedDate: String? = null
)

data class TransactionResponse(
    @SerializedName("ok") val ok: Boolean,
    @SerializedName("transaction") val transaction: Transaction
)

File: sprint1_backend/model/transaction.py
import datetime
from ..app import db, ma



class Transaction(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    usd_amount = db.Column(db.Float, nullable=False)
    lbp_amount = db.Column(db.Float, nullable=False)
    usd_to_lbp = db.Column(db.Boolean, nullable=False)
    added_date = db.Column(db.DateTime)
    user_id = db.Column(db.Integer, db.ForeignKey('user.id'), nullable=True)

    def __init__(self, usd_amount, lbp_amount, usd_to_lbp, user_id=None):
        super(Transaction, self).__init__(
            usd_amount=usd_amount,
            lbp_amount=lbp_amount,
            usd_to_lbp=usd_to_lbp,
            user_id=user_id,
            added_date=datetime.datetime.now()
        )

class TransactionSchema(ma.SQLAlchemyAutoSchema):
    class Meta:
        model = Transaction
        include_fk = True
        fields = ("id", "usd_amount", "lbp_amount", "usd_to_lbp", "added_date", "user_id")
        load_instance = True

transaction_schema = TransactionSchema()
transactions_schema = TransactionSchema(many=True)
